Support for ssl

SonicDMG opened the pull request.

SonicDMG committed:
Adding initial "alpha" support for SSL

SonicDMG committed:
Updated the bootstrap to check if SSL is enabled and configure commands appropriately

SonicDMG committed:
Commit to get Adron to the latest working spot

SonicDMG committed:
Added a simple TODO statement in bootstrap.sh as a reminder for eventual nodesync work

SonicDMG committed:
Updated version from 1.3.0 to 2.1.0 to synchronize versions across all the KillrVideo services with SSL updates

jeffreyscarpenter commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM datastax/dse-server:6.0.0
+FROM datastax/dse-server:6.0.1
```
why not 6.0.2?

SonicDMG replied:
Only 6.0.1 was available at the time I made the change.  If 6.0.2 is available I can change it easily enough, but not tested yet.

SonicDMG replied:
Looks like 6.0.2 is not yet available

jeffreyscarpenter commented on bootstrap.sh:
```diff
@@ -17,6 +18,32 @@ if [ ! -f killrvideo_bootstrapped ]; then
+	# The reference to this file is provided via a volume enabled
+	# on the dse-config container within docker-compose.yaml
+	# in the killrvideo-docker-common repo 
+  	dse_ssl_certfile='/opt/killrvideo-data/cassandra.cert'
```
why killrvideo-data instead of killrvideo-dse-config? Not that it matters, just wondering about consistent naming.

SonicDMG replied:
/opt/killrvideo-data is the path that dse-config uses locally within the Docker container for all of its other files so I am using the same path for consistency.  I did the same thing with all of the other affected services like /opt/killrvideo-generator for the generator.  These paths were already in place before I made any changes.  I could make the change easily enough if you would like.  Will need to retest.

jeffreyscarpenter commented on bootstrap.sh:
```diff
@@ -17,6 +18,32 @@ if [ ! -f killrvideo_bootstrapped ]; then
+  	dse_ssl_certfile='/opt/killrvideo-data/cassandra.cert'
+    	dse_ssl='--ssl'
+
+	# These 2 environment variables are needed for cqlsh to 
```
Why do you mention cqlsh? Oh, I see because it is used below.

jeffreyscarpenter commented on bootstrap.sh:
```diff
@@ -31,9 +58,9 @@ if [ ! -f killrvideo_bootstrapped ]; then
+    cqlsh $dse_ip 9042 -u $admin_user -p $admin_password $dse_ssl $dse_request_timeout -e "CREATE ROLE $KILLRVIDEO_ADMIN_USERNAME with SUPERUSER = true and LOGIN = true and PASSWORD = '$KILLRVIDEO_ADMIN_PASSWORD'"
     # Login as new superuser to delete default superuser (cassandra)
-    cqlsh $dse_ip 9042 -u $KILLRVIDEO_ADMIN_USERNAME -p $KILLRVIDEO_ADMIN_PASSWORD -e "DROP ROLE $admin_user"
+    cqlsh $dse_ip 9042 -u $KILLRVIDEO_ADMIN_USERNAME -p $dse_ssl $dse_request_timeout $KILLRVIDEO_ADMIN_PASSWORD -e "DROP ROLE $admin_user"
```
This looks like a bug: 
-p $dse_ssl $dse_request_timeout $KILLRVIDEO_ADMIN_PASSWORD

Shouldn't it be 
-p $KILLRVIDEO_ADMIN_PASSWORD $dse_ssl $dse_request_timeout

SonicDMG replied:
Oh, nice find.  Let me fix that up. Interestingly, it didn't break. Maybe cqlsh doesn't process $dse_ssl and $dse_request_timeout as the password because they are clearly params and not the value for -p.

jeffreyscarpenter commented on bootstrap.sh:
```diff
@@ -31,9 +58,9 @@ if [ ! -f killrvideo_bootstrapped ]; then
   if [ "$KILLRVIDEO_CREATE_ADMIN_USER" = 'true' ]; then
     echo "=> Creating new superuser $KILLRVIDEO_ADMIN_USERNAME"
-    cqlsh $dse_ip 9042 -u $admin_user -p $admin_password -e "CREATE ROLE $KILLRVIDEO_ADMIN_USERNAME with SUPERUSER = true and LOGIN = true and PASSWORD = '$KILLRVIDEO_ADMIN_PASSWORD'"
+    cqlsh $dse_ip 9042 -u $admin_user -p $admin_password $dse_ssl $dse_request_timeout -e "CREATE ROLE $KILLRVIDEO_ADMIN_USERNAME with SUPERUSER = true and LOGIN = true and PASSWORD = '$KILLRVIDEO_ADMIN_PASSWORD'"
```
These cqlsh commands start to look really messy with all of the options. If we're just repeating the same options over and over, why not just create one variable with all of the options, like "$cqlsh_options" or something like that?

SonicDMG replied:
Because they are both optional, but...I see your point.  Even optional we could still combine them into a single variable. I honestly hadn't thought about doing that as I like to be explicit.

SonicDMG committed:
Reordered params at line 63 per Jeff's bug report in pull request.
The following line: cqlsh $dse_ip 9042 -u $KILLRVIDEO_ADMIN_USERNAME -p $dse_ssl $dse_request_timeout $KILLRVIDEO_ADMIN_PASSWORD -e "DROP ROLE $admin_user" has $dse_ssl and $dse_request_timeout essentially being passed in at the value for "-p". This is now fixed.

Adron reviewed (approved):
LGTM.

Ready to record a 'deploying KillrVideo' video.  :+1:

SonicDMG committed:
Per Jeff's suggestion via https://github.com/KillrVideo/killrvideo-dse-config/pull/1 consolidated multiple variables into one for easier reading

SonicDMG committed:
Upgrade DSE version to 6.0.2 in Dockerfile
